.github/workflows/autodocs.yml

```diff
@@ -0,0 +1,35 @@
+name: Automatic Documentation for Launcher
+
+on:
+  push:
+    branches:
+      - launcher_docs
+
+jobs:
+  update_docs:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    
+    - name: Install Launcher
+      id: install-launcher
+      run: cargo install --git https://github.com/huggingface/text-generation-inference.git text-generation-launcher
+    
+    - name: Update Docs
+      run: |
+        echo text-generation-launcher --help
+        output=$(text-generation-launcher --help)
+        { echo '```'; echo $output; echo '```'; } > launcher_test.md
+        if diff launcher_test.md docs/source/basic_tutorials/launcher.md; then
+          rm docs/source/basic_tutorials/launcher.md
+          { echo '```'; echo $output; echo '```'; } > docs/source/basic_tutorials/launcher.md
+          echo "$(cat docs/source/basic_tutorials/launcher.md)"
+          echo "Docs are updated!"
+          git add docs/source/basic_tutorials/launcher.md
+          git commit -m "Update launcher.md"
+          git push
```

Comment:
I saw in the other PR that there was some discussion about using `git push` inside the action. Was the conclusion that this is safe to do?

Reply:
The concern wasn't this but rather about triggering an infinite loop. Manually running solves this.